src/d3a/models/area/__init__.py

```diff
@@ -332,10 +332,7 @@ def current_balancing_market(self):
             return None
 
     def get_future_market_from_id(self, _id):
-        try:
-            return [m for m in self._markets.markets.values() if m.id == _id][0]
-        except IndexError:
-            return None
+        return self._markets.indexed_future_markets.get(_id, None)
```

Comment:
Ahh, nice! I guess that a dict lookup is less expensive than looping in a list, right?

Reply:
Exactly. Even creating a list takes more time than accessing a dict key.